docs/sources/mimir/configure/configure-ingester-circuit-breakers.md

```diff
@@ -0,0 +1,89 @@
+  Successful trial requests indicate application recovery, and the circuit breaker transitions back to the `closed` state.
+  Failing trial requests indicate that the issues persist, and the circuit breaker transitions back to the `open` state.
+
+## How do Grafana Mimir Ingester circuit breakers work?
+
+A request to a resource protected by a circuit breaker follows these steps:
+
+1. The request tries to acquire a circuit breaker permit.
+1. If the circuit breaker is open, no permit is acquired, and the request fails with a _circuit breaker open error_.
+1. Otherwise, the request acquires a circuit breaker permit and runs.
+1. If the request meets the circuit breaker's failure condition, the circuit breaker records a failure.
+   Otherwise, the circuit breaker records a success.
+
+Depending on the configured frequencies of successes and failures, the circuit breaker transits from one state to another.
+
+{{% admonition type="note" %}}Grafana Mimir ingester circuit breakers are an experimental feature.
+They independently protect Mimir's write and read paths from slow requests.
+They don't protect ingesters from other issues.
+{{% /admonition %}}
+
+More precisely, Grafana Mimir ingesters distinguish between _push requests circuit breakers_ and _read requests circuit breakers_, and you can configure them independently.
+It is possible to configure the maximum allowed duration of both push and read requests, as well as the highest frequency of the slow requests before the circuit breakers open and start protecting the ingesters.
+
+### Push requests circuit breakers
+
+A push request meets the push requests circuit breaker failure condition if its duration is longer than the configured maximum push request duration.
+
+### Read requests circuit breakers
+
+Read requests circuit breakers follow these conditions:
+
+- In order to protect the write path as much as possible, ingesters do not allow read push requests if their push circuit breakers are open.
+  This means that before an ingester tries to acquire a read circuit breaker permit, it first checks if its push circuit breaker is open.
+  If it's open, a circuit breaker open error is returned.
+  Otherwise, the ingester tries to acquire a read requests circuit breaker permit.
+- A read request that acquires a read requests circuit breaker permit meets the read requests circuit breaker failure condition if its duration is longer than the configured read request maximum duration.
+
+## Grafana Mimir Ingester circuit breakers configuration
+
+To enable Grafana Mimir ingester push and read circuit breakers, set `-ingester.push-circuit-breaker.enabled=true` and `-ingester.read-circuit-breaker.enabled=true`.
+You can enable the circuit breakers independently. Enabling one of them does not require enabling the other one.
```

Comment:
any plans to make those on by default in some release? Maybe mention this here if yes

Reply:
I am not sure about this. For the moment circuit breakers are an experimental feature, and they will be disabled by default. In Grafana Cloud they are already enabled, and we are observing their behaviour. Depending on that we will decide how to proceed.